### PR TITLE
[Wallet] Fix issues with escrow payments and withdrawals

### LINKED ISSUE #207: Balance doesn't update after successful escrow withdrawal
### Expected Behavior

Balance should update after escrow withdraws

### Current Behavior

After successful verification, escrow withdraws money but balance doesn't update automatically

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

- Fix balance updating after escrow withdrawal (e.g. after verification)
- Fix balance updating after escrow transfer
- Register standby txs for escrow payments
- Implement smarter checks while fetching latest escrow payments to reduce duplicate requests
- Fix escrow saga error handling 
- Have verification timeout properly trigger verification failure UI
- Silence the celo analytics errors when running debug build
- Remove some dead code

### Tested

Did withdrawal after verification, balance update.
Sent to escrow, standby tx showed correctly and then was removed correctly (see screenshot).
Triggered refresh, pre-retrieved payments were skipped correctly.

### Related issues

- Fixes #207 

### Backwards compatibility

Yes

escrow-standby-tx

### CODE REVIEW

File: packages/mobile/src/escrow/saga.ts [hunk 1]
```diff
@@ -174,56 +198,62 @@ function* reclaimFromEscrow(action: ReclaimPaymentAction) {
...
+  Logger.debug(TAG + '@doFetchSentPayments', 'Fetching valid sent escrowed payments')
+
   try {
-    const escrow = yield call(getEscrowContract, web3)
-    const account = yield call(getConnectedAccount)
-    const recipientsPhoneNumbers = yield select(inviteesSelector)
-    const recipientPhoneNumberToContact = yield select(recipientCacheSelector)
+    const escrow: Escrow = yield call(getEscrowContract, web3)
+    const account: string = yield call(getConnectedAccount)
+    const existingPayments: EscrowedPayment[] = yield select(sentEscrowedPaymentsSelector)
```
Comment (jeanregisser) [inline packages/mobile/src/escrow/saga.ts:222]: Are the explicit types necessary?
This question applies to the other places where explicit types have been added.

File: packages/mobile/src/escrow/saga.ts [hunk 2]
```diff
@@ -37,56 +42,73 @@ import { getConnectedAccount, getConnectedUnlockedAccount } from 'src/web3/saga'
...
+    const escrow: Escrow = yield call(getEscrowContract, web3)
+    const stableToken: StableToken = yield call(getStableTokenContract, web3)
+    const account: string = yield call(getConnectedUnlockedAccount)
 
     Logger.debug(TAG + '@transferToEscrow', 'Approving escrow transfer')
     const convertedAmount = web3.utils.toWei(amount.toString())
     const approvalTx = stableToken.methods.approve(escrow.options.address, convertedAmount)
 
-    yield call(sendTransaction, approvalTx, account, TAG, txId)
+    yield call(sendTransaction, approvalTx, account, TAG, 'approval')
```
Comment (jeanregisser) [inline packages/mobile/src/escrow/saga.ts:56]: Is the last argument just a comment? or does it need to have that specific value for this use case?

**Review by jeanregisser — COMMENTED** [review]
Great work 👏 👍
Just some small clarification questions.

File: packages/mobile/src/escrow/saga.ts (hunk 1 above)
Comment (jmrossy) [reply packages/mobile/src/escrow/saga.ts:222]: The explicit types are necessary for the results of `yield` statements. TS doesn't seem to support that atm

File: packages/mobile/src/escrow/saga.ts (hunk 2 above)
Comment (jmrossy) [reply packages/mobile/src/escrow/saga.ts:56]: The last value can be anything, it's only use is in the logs

File: packages/mobile/src/escrow/saga.ts (hunk 1 above)
Comment (jeanregisser) [reply packages/mobile/src/escrow/saga.ts:222]: Ah yes this is sad 😭 
I hope it will change rather sooner.

**Review by jeanregisser — APPROVED** [review]

File: packages/mobile/src/escrow/saga.ts (hunk 1 above)
Comment (jeanregisser) [reply packages/mobile/src/escrow/saga.ts:222]: Looks like it's coming soon though! https://github.com/redux-saga/redux-saga/pull/1892

**Pull request merged**